packages/mobile/__tests__/pages/Login.test.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { render, fireEvent, wait, waitForElement } from '@testing-library/react-native';
+import configureStore from 'redux-mock-store';
+import AsyncStorage from '@react-native-community/async-storage';
+import { showMessage } from 'react-native-flash-message';
+
+import Wrapper from '../../helpers/Wrapper';
+import { handleLoginFacebook } from '../../src/services/authFacebook';
+import Login, { USER_LOGIN_MUTATION } from '~/pages/Login/index';
+import { createApolloStore } from '../../fixtures/login';
+
+const navigation = {
+  dispatch: jest.fn(),
+  navigate: jest.fn(),
+};
+jest.mock('../../src/services/authFacebook');
+jest.mock('@react-native-community/async-storage');
+jest.mock('react-native-flash-message');
+
+describe('Login', () => {
+  const renderComponent = () => {
+    const mockStore = configureStore();
+    const store = mockStore({});
+    const apolloStore = createApolloStore(USER_LOGIN_MUTATION);
+
+    const component = (
+      <Wrapper apolloStore={apolloStore} reduxStore={store}>
+        <Login navigation={navigation} />
+      </Wrapper>
+    );
+
+    return render(component);
+  };
+
+  it('shows loading when press login button', async () => {
+    const { getByText, getByTestId } = renderComponent();
+
+    const btn = getByText('Entrar com facebook');
+
+    fireEvent.press(btn);
+
+    await waitForElement(() => getByTestId('loading'));
+  });
+
+  describe('when login is successful', () => {
+    beforeEach(() => {
+      handleLoginFacebook.mockImplementation(() => ({
+        error: false,
+        user: true,
+      }));
+    });
+
+    it('handle properly the login flow', async () => {
+      const { getByText } = renderComponent();
+
+      const btn = getByText('Entrar com facebook');
+
+      fireEvent.press(btn);
+
+      await wait();
```

Comment:
What is the purpose of this empty `wait`? 🤔

Reply:
The purpose of this is await the mock promises has been resolved.

Reference: https://www.native-testing-library.com/docs/api-async